### PR TITLE
Add micro-benchmark support functions

### PR DESCRIPTION
**Pull request by mattkretz**
ChangeLog:

	* .gitignore: Ignore generated test.S.
	* Makefile: Optimize test.cpp build and store asm output for reference.
	* README.md: Document new header.
	* vir/simd_benchmarking.h: New file.
	* vir/test.cpp: Test fake_modify and fake_read.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+test.S
```
Comment (ChristianTackeGSI) [inline .gitignore:1]: I have learned to like the "build"-directory style of things.

So mid-term, I would suggest to put build-time artefacts in some `build/` directory and only add that to gitignore.

File: vir/simd_benchmarking.h [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	  const auto& d = stdx::__data(x);
+	  if constexpr (sizeof(d) <= 8)
+	    asm volatile("" ::"g"(d));
+	  else if constexpr (requires {{d._M_data};})
+	    asm volatile("" ::"x"(d));
+	  else if constexpr (requires {{d._S_tuple_size};})
+	    stdx::__for_each(d, [](auto, const auto& element) {
+	      fake_read1(element);
+	    });
+	  else
```
Comment (ChristianTackeGSI) [inline vir/simd_benchmarking.h:73]: Although this only one command inside the block, it spans more than one line. I would really prefer `{,}`-braces around this block.

File: .gitignore (hunk 1 above)
Comment (mattkretz) [reply .gitignore:1]: Yes. As long as it's one generated file vs. one generated directory. I'm not ready to add complexity. Keep it simple for now.

File: vir/simd_benchmarking.h (hunk 2 above)
Comment (mattkretz) [reply vir/simd_benchmarking.h:73]: I prefer it without the braces. The braces would add two more lines, making the function longer and thus harder to capture in one go.

**Pull request merged**